Implementacion inicial del Simon-dice

nicolasmail209 opened the pull request.

nicolasmail209 committed:
Crea un primer intento

nicolasmail209 committed:
Borra comentarios de mas en el primer intento

nicolasmail209 committed:
Habilita todos los cuadrados

Me habia olvidado de habilitar los cuadrados azul y amarillo.

fsodano commented on index.html:
```diff
@@ -0,0 +1,26 @@
+    
+    <div style="height: 100px; width: 100px; background-color: rgb(138, 2, 2)" id="dos"></div>
+    
+    <div style="height: 100px; width: 100px; background-color: rgb(2, 2, 129)" id="tres"></div>
```
evitá los style, sacalos a clases y reusá esas clases

fsodano commented on index.js:
```diff
@@ -0,0 +1,195 @@
+    const $cuadrado = document.querySelector("#" + id);
+
+    if(id === "uno"){
+        $cuadrado.style.backgroundColor = "rgb(4, 252, 4)";
```
mejor cambiarle el nombre de la clase, por ejemplo podrías tener declaradas clases en CSS onda `#uno.encendido` entonces toda esta función sólo se termina haciendo `$cuadrado.classList.add("encendido")`

en tu CSS tenés entonces

```
#uno.encendido{
  background-color:  rgb(1, 112, 1);
}

#dos.encendido{
  background-color:  rgb(241, 3, 3);
}
```

etc.

después en el setTimeout (hacés uno solo) y lo que hacés es el `.remove('encendido')` y listo.

fsodano commented on index.js:
```diff
@@ -0,0 +1,195 @@
+    
+    const secuencia = [];
+
+    function getRandomIntInclusive(min, max) {
+        min = Math.ceil(min);
+        max = Math.floor(max);
+        return Math.floor(Math.random() * (max - min + 1) + min);
+    }
```
esta función no va acá, va afuera; así como está ahora cada vez que llamás a crearSecuencia estás creando la función getRandomIntInclusive.

Y ojo con el spanglish, ponele `obtenerNumeroAleatorioInclusive`

fsodano commented on index.js:
```diff
@@ -0,0 +1,195 @@
+}
+
+
+function transformarSecuenciaAPalabras(){
```
no es más fácil `["cero", "uno", "dos", "tres", "cuatro"]` y listo?

fsodano commented on index.js:
```diff
@@ -0,0 +1,195 @@
+   encender(secuencia[counter]);
+   counter+=1;
+   if (counter == (secuencia.length)) { clearInterval(interval); }
+}, 3000);
```
este 3000 es un número mágico, sacalo a una constante.

fsodano commented on index.js:
```diff
@@ -0,0 +1,195 @@
+
+    
+
+    const $div1 = document.querySelector("#uno");
+    const $div2 = document.querySelector("#dos");
+    const $div3 = document.querySelector("#tres");
+    const $div4 = document.querySelector("#cuatro");
```
no le pongas `$div` porque si el día de mañana es un span no tenés que andar cambiándole el nombre, ponele lo que representa onda `$cuadrado`

fsodano commented on index.js:
```diff
@@ -0,0 +1,195 @@
+    const $div3 = document.querySelector("#tres");
+    const $div4 = document.querySelector("#cuatro");
+
+    $div1.onclick = function(){
```
todos los divs hacen lo mismo, este copy paste de 4 cuadrados está mal, rehacelo para que lo puedas resolver con una única función genérica.

nicolasmail209 committed:
Se regresa al momento anterior a hacer las correcciones pedidas en la PR

Si hubiera problemas mas adelante, en vez de comenzar desde cero, se empezar desde aqui.

nicolasmail209 committed:
Todo lo que estaba en el atributo Style pasa a un archivo CSS

Asi se puede reutilizar la clase .cuadrado

nicolasmail209 committed:
Cambia la forma de encender los cuadrados

Nos ahorramos muchas lineas de codigo.

nicolasmail209 committed:
Se cambia el nombre y la ubicacion de la funcion getRandomIntInclusive()

Asi la funcion solo se crea una vez.

nicolasmail209 committed:
Se elimina una funcion que no tenia sentido

Asi logramos lo mismo en menos pasos.

nicolasmail209 committed:
Se transforma un numero magico en una constante

Asi se sabe que significa.

nicolasmail209 committed:
Se cambia el nombre a los cuadrados y se crea una funcion generica para reducir el codigo

Asi el nombre de los cuadrados sera relevante por mas que su tag cambie, y con la funcion generica se reducen las lineas de codigo innecesario.

nicolasmail209 committed:
Se formatean todos los archivos con Prettier

Asi es mas facil su lectura.
